hr_timesheet_purchase_order/models/hr_timesheet_recurrence.py

```diff
@@ -0,0 +1,362 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from calendar import monthrange
+
+from dateutil.relativedelta import relativedelta
+from dateutil.rrule import (
+    DAILY,
+    FR,
+    MO,
+    MONTHLY,
+    SA,
+    SU,
+    TH,
+    TU,
+    WE,
+    WEEKLY,
+    YEARLY,
+    rrule,
+)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+MONTHS = {
+    "january": 31,
+    "february": 28,
+    "march": 31,
+    "april": 30,
+    "may": 31,
+    "june": 30,
+    "july": 31,
+    "august": 31,
+    "september": 30,
+    "october": 31,
+    "november": 30,
+    "december": 31,
+}
+
+DAYS = {
+    "mon": MO,
+    "tue": TU,
+    "wed": WE,
+    "thu": TH,
+    "fri": FR,
+    "sat": SA,
+    "sun": SU,
+}
+
+WEEKS = {
+    "first": 1,
+    "second": 2,
+    "third": 3,
+    "last": 4,
+}
+
+
+class HRTimeSheetRecurrence(models.Model):
```

Comment:
Please add docstring for each method with describe arguments.

Reply:
okay, I'll add it